package-name-map/src/test/package-name-map_test.ts

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chai = require('chai');
+
+import {PackageNameMap, isPathSegmentPrefix} from '../package-name-map.js';
+
+const {assert} = chai;
+
+suite('PackageNameMap', () => {
+  suite('resolve', () => {
+    const referrerURL = 'http://foo.com/';
+
+    suite('does not modify already valid specifiers', () => {
+      const map = new PackageNameMap({
+        packages: {
+          foo: {
+            main: 'index.js',
+          },
+        },
+      });
+
+      test('does not modify URLs', () => {
+        assert.equal(
+          map.resolve('https://bar.com', referrerURL),
+          'https://bar.com'
+        );
+      });
+
+      test('does not modify valid paths', () => {
+        assert.equal(map.resolve('/foo', referrerURL), '/foo');
+        assert.equal(map.resolve('./foo', referrerURL), './foo');
+        assert.equal(map.resolve('../foo', referrerURL), '../foo');
+      });
+    });
+
+    suite('top-level package names', () => {
+      const map = new PackageNameMap({
+        packages: {
+          app: {
+            main: 'src/index.js',
+          },
+          lodash: {
+            path: '/node_modules/lodash-es',
+            main: 'lodash.js',
+          },
+          '@polymer/polymer': {
+            path: '/node_modules/@polymer/polymer',
+            main: 'polymer.js',
+          },
+          '@polymer/polymer-foo': {
+            path: '/node_modules/@polymer/polymer-foo',
+            main: 'polymer-foo.js',
+          },
+        },
+      });
+
+      test('resolves package name for a package with only a main', () => {
+        assert.equal(map.resolve('app', referrerURL), 'app/src/index.js');
```

Comment:
The result should be `http://foo.com/app/src/index.js`; we want `resolve()` to always return an absolute URL.

Reply:
done